### ISSUE TITLE
SetPagingItem not called when tapped (not scrolled).

### ISSUE THREAD

**Issue by glennposadas** [rechsteiner/Parchment#251]
Hi, just a minor issue. I have a custom PagingCell and this method isn't called when the title is tapped and not scrolled.

`override func setPagingItem(_ pagingItem: PagingItem, selected: Bool, options: PagingOptions) { }
`
So what happens is that the tapped title does not get highlighted and the last scrolled title or last selected title stays highlighted though it's not selected anymore (and the indicator is not below that last selected title anymore). I hope my explanation is clear enough :) Starred this control, btw.

Demo:
![alt text](http://i67.tinypic.com/2eo87t4.gif "Bug")

Thank you!

**Comment by glennposadas** [comment]
GIF/Demo: https://im4.ezgif.com/tmp/ezgif-4-55ec7e1c6c.gif

**Comment by glennposadas** [comment]
My colleague has fixed this issue. He adjusted the threshold, like so:

// Thresholds to update view layouts call delegates
            if (progress >= 0.99 && self.afterViewController != nil) {
                self.didFinishScrolling(to: self.afterViewController!)
                scrollView.setContentOffset(CGPoint(x: self.view.bounds.width, y: 0), animated: true)
            } else if (progress <= -0.99  && self.beforeViewController != nil) {
                self.didFinishScrolling(to: self.beforeViewController!)
                scrollView.setContentOffset(CGPoint(x: self.view.bounds.width, y: 0), animated: true)
            } else if (progress == 0  && self.selectedViewController != nil) {
                self.didFinishScrolling(to: self.selectedViewController!)
            }

I'll keep this open though, for the contributors to notice this.

**Comment by rechsteiner** [comment]
Thanks for the report! Will look into getting this fixed 🙂

**Comment by rechsteiner** [comment]
@glennposadas I looked a bit more into this and I'm not able to reproduce the issue. The `setPagingItem` method gets called both when an item is tapped and after a scroll has finished.

If you need to update the cell _while_ the user is scrolling, you need to override `apply(layoutAttributes:)`. E.g: https://github.com/rechsteiner/Parchment/blob/master/Parchment/Classes/PagingTitleCell.swift#L64

 Are you sure this is an issue with Parchment? If so, could you provide some ways of reproducing this?

**Comment by glennposadas** [comment]
Thanks @rechsteiner for taking time to check this issue out. I believe it has to do with how I'm using the Parchment. The layout of my controller is a bit complicated, with lots of child controllers. We can close this for now since I'm the only one whos experiencing such an issue.

**Comment by beilusen** [comment]
I had a similar problem. When I used a button to trigger select (pagingItem: PagingItem, animated: Bool = false), I printed progress in the isScrollingFromItem currentPagingItem of the delegate and found that it did not become 1 or -1, but 0.99. As a result, item has no selection state, so it is not possible to call serelct again. Forgive me for poor english
![截屏2020-03-2115 23 01](https://user-images.githubusercontent.com/22909657/77221802-fdbcfe80-6b87-11ea-8c6f-4c61cd980f40.png)
